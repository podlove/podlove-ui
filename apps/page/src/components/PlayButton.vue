<template>
  <play-button
    class="play-button"
    :size="size"
    @play="play"
    @pause="pause"
    :type="state.playing ? 'pause' : 'play'"
  />
</template>

<script setup lang="ts">
import { mapState, injectStore } from 'redux-vuex';
import { PlayButton } from '@podlove/components';
import { selectors, actions } from '../logic';

const props = defineProps<{ id: string; size: number }>();

const store = injectStore();

const state = mapState({
  playing: selectors.episode.playing(props.id)
});

const play = () => {
  store.dispatch(actions.player.playEpisode({ id: props.id }))
}

const pause = () => {
  store.dispatch(actions.player.pauseEpisode())
}
</script>

<style scoped>
.play-button {
  --podlove-component--play-button--text-color: rgb(var(--primary-color-200));
  --podlove-component--play-button--background: rgb(var(--primary-color-800));
}
</style>
