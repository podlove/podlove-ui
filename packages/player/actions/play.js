import { createAction } from './action-creator'

import {
  REQUEST_PLAY,
  REQUEST_PAUSE,
  REQUEST_RESTART,
  BACKEND_PLAY,
  BACKEND_PAUSE,
  BACKEND_END,
  BACKEND_LOADING_START,
  BACKEND_LOADING_END,
  BACKEND_BUFFER,
  BACKEND_ERROR
} from './types'

export const requestPlay = createAction(REQUEST_PLAY)
export const backendPlay = createAction(BACKEND_PLAY)

export const requestPause = createAction(REQUEST_PAUSE)
export const backendPause = createAction(BACKEND_PAUSE)

export const requestRestart = createAction(REQUEST_RESTART)

export const backendLoadingStart = createAction(BACKEND_LOADING_START)
export const backendLoadingEnd = createAction(BACKEND_LOADING_END)
export const backendBuffer = createAction(BACKEND_BUFFER)
export const backendEnd = createAction(BACKEND_END)

export const backendError = createAction(BACKEND_ERROR)
