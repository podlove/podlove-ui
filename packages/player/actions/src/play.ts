import { createAction } from 'redux-actions';
import type { PodloveWebPlayerAudioError, PodloveWebPlayerAudioProps } from '@podlove/types';

import {
  REQUEST_PLAY,
  REQUEST_PAUSE,
  REQUEST_RESTART,
  BACKEND_PLAY,
  BACKEND_PAUSE,
  BACKEND_END,
  BACKEND_LOADING_START,
  BACKEND_LOADING_END,
  BACKEND_BUFFER,
  BACKEND_ERROR,
  REQUEST_LOAD,
  REQUEST_STOP
} from './types';

export type requestPlayPayload = void;
export type backendPlayPayload = void;
export type requestPausePayload = void;
export type requestLoadPayload = void;
export type backendPausePayload = void;
export type requestStopPayload = void;
export type requestRestartPayload = void;
export type backendLoadingStartPayload = void;
export type backendLoadingEndPayload = PodloveWebPlayerAudioProps;
export type backendBufferPayload = [number, number][];
export type backendEndPayload = void;
export type backendErrorPayload = PodloveWebPlayerAudioError;

export const requestPlay = createAction<requestPlayPayload>(REQUEST_PLAY);
export const backendPlay = createAction<backendPlayPayload>(BACKEND_PLAY);

export const requestPause = createAction<requestPausePayload>(REQUEST_PAUSE);
export const requestStop = createAction<requestStopPayload>(REQUEST_STOP);
export const requestLoad = createAction<requestLoadPayload>(REQUEST_LOAD);
export const backendPause = createAction<backendPausePayload>(BACKEND_PAUSE);

export const requestRestart = createAction<requestRestartPayload>(REQUEST_RESTART);

export const backendLoadingStart = createAction<backendLoadingStartPayload>(BACKEND_LOADING_START);
export const backendLoadingEnd = createAction<backendLoadingEndPayload>(BACKEND_LOADING_END);
export const backendBuffer = createAction<backendBufferPayload>(BACKEND_BUFFER);
export const backendEnd = createAction<backendEndPayload>(BACKEND_END);

export const backendError = createAction<backendErrorPayload>(BACKEND_ERROR);
