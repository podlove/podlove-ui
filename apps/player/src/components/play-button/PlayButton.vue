<template>
  <play-button
    class="podlove-player--play-button"
    :class="{ hover}"
    v-if="button.type"
    :type="button.type"
    :title="button.a11y"
    :size="size"
    :label="button.label"
    data-test="play-button"
    @play="dispatch"
    @pause="dispatch"
    @restart="dispatch"
    @mouseover="mouseOver()"
    @mouseleave="mouseLeave()"
  />
</template>

<script lang="ts" setup>
import { ref, computed } from 'vue';
import { toHumanTime } from '@podlove/utils/time';
import { mapState, injectStore } from 'redux-vuex';
import { PlayButton } from '@podlove/components';
import { useI18n } from 'vue-i18n';

import select from '../../store/selectors/index.js';

const { t } = useI18n();

const props = defineProps({
  label: {
    type: String,
    default: null
  },
  size: {
    type: Number,
    default: 50
  },
  variant: {
    type: String,
    validate: (val) => ['details', 'simple'].includes(val),
    default: 'details'
  }
});

const hover = ref(false);

const state = mapState({
  playButton: select.components.playButton,
  duration: select.duration,
  playtime: select.playtime,
  pauseA11y: select.accessibility.playButtonPause,
  durationA11y: select.accessibility.playButtonDuration,
  replayA11y: select.accessibility.playButtonReplay,
  playA11y: select.accessibility.playButtonPlay,
  errorA11y: select.accessibility.playButtonError
});

const dispatch = injectStore().dispatch;

const button = computed(() => {
  switch (state.playButton) {
    case 'paused':
      return {
        type: 'play',
        a11y: t(state.playA11y.key, state.playA11y.attr),
        width: '50px'
      };
    default:
    case 'duration':
      return {
        type: 'play',
        a11y: t(state.durationA11y.key, state.durationA11y.attr),
        label:
          props.variant === 'details'
            ? props.label || toHumanTime(state.playtime > 0 ? state.playtime : state.duration)
            : null
      };
    case 'remaining':
      return {
        type: 'play',
        label: props.variant === 'details' ? toHumanTime(state.playtime) : null
      };
    case 'replay':
      return {
        type: 'restart',
        a11y: t(state.replayA11y.key, state.replayA11y.attr),
        label: t('PLAYER.REPLAY')
      };
    case 'loading':
      return {
        type: 'loading'
      };
    case 'playing':
      return {
        type: 'pause',
        a11y: t(state.pauseA11y.key, state.pauseA11y.attr)
      };
    case 'error':
      return {
        type: 'restart',
        a11y: t(state.errorA11y.key, state.errorA11y.attr)
      };
    case 'retry':
      return {
        type: 'restart',
        label: t('PLAYER.RETRY')
      };
    case 'end':
      return {
        type: 'restart'
      };
  }
});

const mouseOver = () => {
  hover.value = true;
};

const mouseLeave = () => {
  hover.value = false;
};
</script>

<style lang="postcss" scoped>
.podlove-player--play-button {
  --podlove-component--play-button--text-color: var(--podlove-player--play-button--color);
  --podlove-component--play-button--background: var(--podlove-player--play-button--background);
}

.podlove-player--play-button.hover {
  --podlove-component--play-button--text-color: var(--podlove-player--play-button--color-hover);
}
</style>
