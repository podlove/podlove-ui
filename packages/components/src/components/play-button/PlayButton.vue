<script lang="ts" setup>
import { requestPlay, requestPause, requestRestart } from '@podlove/player-actions/play';
import { computed } from 'vue';

import Play from './states/Play.vue';
import Pause from './states/Pause.vue';
import Loading from './states/Loading.vue';
import Restart from './states/Restart.vue';

const types: { [key: string]: any } = {
  play: Play,
  pause: Pause,
  loading: Loading,
  restart: Restart
}

const props = defineProps({
  type: {
    type: String,
    required: true,
    validator: (val: string) => ['play', 'pause', 'loading', 'restart'].includes(val)
  },
  label: {
    type: String,
    default: ''
  },
  size: {
    type: Number,
    default: 50
  }
});

const emit = defineEmits(['play', 'pause', 'restart']);

const clickHandler = () => {
  switch (props.type) {
    case 'play':
      emit('play', requestPlay());
      break;
    case 'pause':
      emit('pause', requestPause());
      break;
    case 'restart':
      emit('restart', requestRestart());
      break;
  }
};

const styleSize = computed(() => `${props.size}px`);
const maxWidth = computed(() => props.label && props.type !== 'loading' ? '150px' : `${props.size}px`);
</script>

<template>
  <button
    :id="`play-button--${type}`"
    ref="playbutton"
    class="podlove-component-play-button"
    @click="clickHandler()"
  >
    <div class="play-button--wrapper flex items-center justify-center overflow-hidden rounded-full">
      <transition name="component" mode="out-in">
        <component
          :is="types[type]"
          :id="`play-button--${type}`"
          ref="component"
          class="component py-0"
          :size="size / 2"
          :class="{
            'pl-4 pr-5': label && type !== 'loading'
          }"
        >
          <span
            v-if="label"
            data-test="play-button--label"
            class="play-button--label truncate ml-2 text-base font-normal font-variant-numeric"
          >
            {{ label }}
          </span>
        </component>
      </transition>
    </div>
  </button>
</template>

<style lang="postcss" scoped>
.podlove-component-play-button {
  --podlove-component--icon--color: var(
    --podlove-component--play-button--text-color,
    var(--podlove-components-text)
  );
}

.podlove-component-play-button .play-button--wrapper {
  background-color: var(
    --podlove-component--play-button--background,
    var(--podlove-components-background)
  );
  height: v-bind('styleSize');
  min-width: v-bind('styleSize');
  border-radius: calc(v-bind('styleSize') / 2);
  max-width: v-bind('maxWidth');
}

.podlove-component-play-button .play-button--label {
  color: var(--podlove-component--play-button--text-color, var(--podlove-components-text));
}

.component-enter-active,
.component-leave-active {
  transition: opacity 300ms ease;
}

.component-enter,
.component-leave-to {
  opacity: 0;
}
</style>
