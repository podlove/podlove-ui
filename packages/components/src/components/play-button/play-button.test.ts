import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { requestPlay, requestPause, requestRestart } from '@podlove/player-actions/play';
import PlayButton from './PlayButton.vue';

describe('PlayButton', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(PlayButton, {
        props: {
          type: 'play'
        }
      });

      expect(wrapper.vm).toBeTruthy();
    });
  });

  describe('props', () => {
    const types = ['play', 'pause', 'loading', 'restart'];
    const props = [
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'color',
        value: '#fff'
      },
      {
        prop: 'label',
        value: 'test-label'
      }
    ];

    types.forEach((type) => {
      props.forEach(({ prop, value }) => {
        test(`should render type '${type}' with '${prop}'`, async () => {
          const wrapper = await mount(PlayButton, {
            props: {
              type,
              [prop]: value
            }
          });

          expect(wrapper.element).toMatchSnapshot();
        });
      });
    });
  });

  describe('events', () => {
    test('should dispatch REQUEST_PLAY on click with type play', async () => {
      const wrapper = await mount(PlayButton, {
        props: {
          type: 'play'
        }
      });

      wrapper.trigger('click');
      expect(wrapper.emitted().play).toEqual([[requestPlay()]]);
    });

    test('should dispatch REQUEST_PAUSE on click with type play', async () => {
      const wrapper = await mount(PlayButton, {
        props: {
          type: 'pause'
        }
      });

      wrapper.trigger('click');
      expect(wrapper.emitted().pause).toEqual([[requestPause()]]);
    });

    test('should dispatch REQUEST_PLAY on click with type play', async () => {
      const wrapper = await mount(PlayButton, {
        props: {
          type: 'restart'
        }
      });

      wrapper.trigger('click');
      expect(wrapper.emitted().restart).toEqual([[requestRestart()]]);
    });
  });
});
