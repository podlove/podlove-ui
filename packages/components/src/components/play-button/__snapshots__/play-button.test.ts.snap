// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayButton > props > should render type 'loading' with 'background' 1`] = `
<button
  background="#fff"
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--loading"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <div
        aria-hidden="true"
        class="spinner component py-0"
        data-v-03d7d96a=""
        data-v-825025bb=""
        id="play-button--loading"
      >
        <div
          class="double-bounce1"
          data-v-825025bb=""
        />
        <div
          class="double-bounce2"
          data-v-825025bb=""
        />
      </div>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'loading' with 'color' 1`] = `
<button
  class="podlove-component-play-button"
  color="#fff"
  data-v-03d7d96a=""
  id="play-button--loading"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <div
        aria-hidden="true"
        class="spinner component py-0"
        data-v-03d7d96a=""
        data-v-825025bb=""
        id="play-button--loading"
      >
        <div
          class="double-bounce1"
          data-v-825025bb=""
        />
        <div
          class="double-bounce2"
          data-v-825025bb=""
        />
      </div>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'loading' with 'label' 1`] = `
<button
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--loading"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <div
        aria-hidden="true"
        class="spinner component py-0"
        data-v-03d7d96a=""
        data-v-825025bb=""
        id="play-button--loading"
      >
        <div
          class="double-bounce1"
          data-v-825025bb=""
        />
        <div
          class="double-bounce2"
          data-v-825025bb=""
        />
      </div>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'pause' with 'background' 1`] = `
<button
  background="#fff"
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--pause"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0"
        data-v-03d7d96a=""
        id="play-button--pause"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <rect
            fill="currentColor"
            height="16"
            rx="1.5"
            width="3"
            x="7"
            y="4"
          />
          <rect
            fill="currentColor"
            height="16"
            rx="1.5"
            width="3"
            x="14"
            y="4"
          />
          
        </svg>
        
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'pause' with 'color' 1`] = `
<button
  class="podlove-component-play-button"
  color="#fff"
  data-v-03d7d96a=""
  id="play-button--pause"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0"
        data-v-03d7d96a=""
        id="play-button--pause"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <rect
            fill="currentColor"
            height="16"
            rx="1.5"
            width="3"
            x="7"
            y="4"
          />
          <rect
            fill="currentColor"
            height="16"
            rx="1.5"
            width="3"
            x="14"
            y="4"
          />
          
        </svg>
        
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'pause' with 'label' 1`] = `
<button
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--pause"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0 pl-4 pr-5"
        data-v-03d7d96a=""
        id="play-button--pause"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <rect
            fill="currentColor"
            height="16"
            rx="1.5"
            width="3"
            x="7"
            y="4"
          />
          <rect
            fill="currentColor"
            height="16"
            rx="1.5"
            width="3"
            x="14"
            y="4"
          />
          
        </svg>
        
        <span
          class="play-button--label truncate ml-2 text-base font-normal font-variant-numeric"
          data-test="play-button--label"
          data-v-03d7d96a=""
        >
          test-label
        </span>
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'play' with 'background' 1`] = `
<button
  background="#fff"
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--play"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0"
        data-v-03d7d96a=""
        id="play-button--play"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M6 5.76341C6 5.19411 6.60936 4.83238 7.10914 5.10498L18.5429 11.3416C19.064 11.6258 19.064 12.3742 18.5429 12.6584L7.10914 18.895C6.60936 19.1676 6 18.8059 6 18.2366V5.76341Z"
            fill="currentColor"
          />
          
        </svg>
        
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'play' with 'color' 1`] = `
<button
  class="podlove-component-play-button"
  color="#fff"
  data-v-03d7d96a=""
  id="play-button--play"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0"
        data-v-03d7d96a=""
        id="play-button--play"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M6 5.76341C6 5.19411 6.60936 4.83238 7.10914 5.10498L18.5429 11.3416C19.064 11.6258 19.064 12.3742 18.5429 12.6584L7.10914 18.895C6.60936 19.1676 6 18.8059 6 18.2366V5.76341Z"
            fill="currentColor"
          />
          
        </svg>
        
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'play' with 'label' 1`] = `
<button
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--play"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0 pl-4 pr-5"
        data-v-03d7d96a=""
        id="play-button--play"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M6 5.76341C6 5.19411 6.60936 4.83238 7.10914 5.10498L18.5429 11.3416C19.064 11.6258 19.064 12.3742 18.5429 12.6584L7.10914 18.895C6.60936 19.1676 6 18.8059 6 18.2366V5.76341Z"
            fill="currentColor"
          />
          
        </svg>
        
        <span
          class="play-button--label truncate ml-2 text-base font-normal font-variant-numeric"
          data-test="play-button--label"
          data-v-03d7d96a=""
        >
          test-label
        </span>
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'restart' with 'background' 1`] = `
<button
  background="#fff"
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--restart"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0"
        data-v-03d7d96a=""
        id="play-button--restart"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M23 12c0 1.042-.154 2.045-.425 3h-2.101c.335-.94.526-1.947.526-3 0-4.962-4.037-9-9-9-1.706 0-3.296.484-4.655 1.314l1.858 2.686h-6.994l2.152-7 1.849 2.673c1.684-1.049 3.659-1.673 5.79-1.673 6.074 0 11 4.925 11 11zm-6.354 7.692c-1.357.826-2.944 1.308-4.646 1.308-4.962 0-9-4.038-9-9 0-1.053.191-2.06.525-3h-2.1c-.271.955-.425 1.958-.425 3 0 6.075 4.925 11 11 11 2.127 0 4.099-.621 5.78-1.667l1.853 2.667 2.152-6.989h-6.994l1.855 2.681z"
            fill="currentColor"
          />
          
        </svg>
        
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'restart' with 'color' 1`] = `
<button
  class="podlove-component-play-button"
  color="#fff"
  data-v-03d7d96a=""
  id="play-button--restart"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0"
        data-v-03d7d96a=""
        id="play-button--restart"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M23 12c0 1.042-.154 2.045-.425 3h-2.101c.335-.94.526-1.947.526-3 0-4.962-4.037-9-9-9-1.706 0-3.296.484-4.655 1.314l1.858 2.686h-6.994l2.152-7 1.849 2.673c1.684-1.049 3.659-1.673 5.79-1.673 6.074 0 11 4.925 11 11zm-6.354 7.692c-1.357.826-2.944 1.308-4.646 1.308-4.962 0-9-4.038-9-9 0-1.053.191-2.06.525-3h-2.1c-.271.955-.425 1.958-.425 3 0 6.075 4.925 11 11 11 2.127 0 4.099-.621 5.78-1.667l1.853 2.667 2.152-6.989h-6.994l1.855 2.681z"
            fill="currentColor"
          />
          
        </svg>
        
        
      </span>
    </transition-stub>
  </div>
</button>
`;

exports[`PlayButton > props > should render type 'restart' with 'label' 1`] = `
<button
  class="podlove-component-play-button"
  data-v-03d7d96a=""
  id="play-button--restart"
>
  <div
    class="play-button--wrapper flex items-center justify-center overflow-hidden"
    data-v-03d7d96a=""
  >
    <transition-stub
      appear="false"
      css="true"
      data-v-03d7d96a=""
      mode="out-in"
      name="component"
      persisted="false"
    >
      <span
        aria-hidden="true"
        class="flex justify-center items-center component py-0 pl-4 pr-5"
        data-v-03d7d96a=""
        id="play-button--restart"
      >
        <svg
          aria-hidden="true"
          class="podlove-component-icon"
          data-v-258dccf6=""
          fill="none"
          height="25"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M23 12c0 1.042-.154 2.045-.425 3h-2.101c.335-.94.526-1.947.526-3 0-4.962-4.037-9-9-9-1.706 0-3.296.484-4.655 1.314l1.858 2.686h-6.994l2.152-7 1.849 2.673c1.684-1.049 3.659-1.673 5.79-1.673 6.074 0 11 4.925 11 11zm-6.354 7.692c-1.357.826-2.944 1.308-4.646 1.308-4.962 0-9-4.038-9-9 0-1.053.191-2.06.525-3h-2.1c-.271.955-.425 1.958-.425 3 0 6.075 4.925 11 11 11 2.127 0 4.099-.621 5.78-1.667l1.853 2.667 2.152-6.989h-6.994l1.855 2.681z"
            fill="currentColor"
          />
          
        </svg>
        
        <span
          class="play-button--label truncate ml-2 text-base font-normal font-variant-numeric"
          data-test="play-button--label"
          data-v-03d7d96a=""
        >
          test-label
        </span>
        
      </span>
    </transition-stub>
  </div>
</button>
`;
